src/objectid.ts

```diff
@@ -72,7 +72,7 @@ export class ObjectId {
       // Generate a new id
       this[kId] = ObjectId.generate(typeof workingId === 'number' ? workingId : undefined);
     } else if (ArrayBuffer.isView(workingId) && workingId.byteLength === 12) {
-      this[kId] = ensureBuffer(workingId);
+      this[kId] = workingId instanceof Buffer ? workingId : ensureBuffer(workingId);
```

Comment:
Is it intentional that this check is added both here and inside of `ensureBuffer`?

Reply:
Yea with my intention being to avoid the overhead of even calling `ensureBuffer`